src/main/java/aztech/modern_industrialization/machines/multiblocks/MultiblockMachineBlockEntity.java

```diff
@@ -33,6 +33,8 @@
 import net.minecraft.world.phys.BlockHitResult;
 
 public abstract class MultiblockMachineBlockEntity extends MachineBlockEntity {
+    protected ShapeMatcher shapeMatcher;
```

Comment:
Make this nullable and keep it private, IMO.

Reply:
This needs to be accessible in subclasses to check for success/failure or to get the matched hatches in onRematch